tensorvis/tensorvisbase/src/processors/tensorfield3dmetadata.cpp

```diff
@@ -29,510 +29,104 @@
-        }
-
-        tensorFieldOut_->addMetaData<ShapeFactor>(shapeFactor, TensorFeature::ShapeFactor);
-    }
-    if (isotropicScaling_.get() && !tensorFieldOut_->hasMetaData<IsotropicScaling>()) {
-        std::vector<double> isotropicScaling;
-
-        const auto& majorEigenValues = tensorFieldOut_->getMetaData<MajorEigenValues>();
-        const auto& intermediateEigenValues =
-            tensorFieldOut_->getMetaData<IntermediateEigenValues>();
-        const auto& minorEigenValues = tensorFieldOut_->getMetaData<MinorEigenValues>();
-
-        for (size_t i = 0; i < majorEigenValues.size(); i++) {
-            isotropicScaling.emplace_back(
-                (majorEigenValues[i] + intermediateEigenValues[i] + minorEigenValues[i]) / 3.0);
-        }
+void TensorField3DMetaData::initializeResources() {
+    if (!inport_.hasData()) return;
 
-        tensorFieldOut_->addMetaData<IsotropicScaling>(isotropicScaling,
-                                                       TensorFeature::IsotropicScaling);
-    }
-    if (rotation_.get() && !tensorFieldOut_->hasMetaData<Rotation>()) {
-        std::vector<double> rotation;
+    NetworkLock l;
 
-        rotation.resize(tensors.size(), 0.0);
+    auto comp = getPropertiesByType<CompositeProperty>().front();
+    auto props = comp->getProperties();
 
-        tensorFieldOut_->addMetaData<Rotation>(rotation, TensorFeature::Rotation);
+    for (auto prop : props) {
+        comp->removeProperty(prop);
     }
-    if (frobeniusNorm_.get() && !tensorFieldOut_->hasMetaData<FrobeniusNorm>()) {
-        std::vector<double> frobeniusNorm;
 
-        const auto& majorEigenValues = tensorFieldOut_->getMetaData<MajorEigenValues>();
-        const auto& intermediateEigenValues =
-            tensorFieldOut_->getMetaData<IntermediateEigenValues>();
-        const auto& minorEigenValues = tensorFieldOut_->getMetaData<MinorEigenValues>();
-
-        for (size_t i = 0; i < majorEigenValues.size(); i++) {
-            auto eigenValues = std::array<double, 3>{
-                majorEigenValues[i], intermediateEigenValues[i], minorEigenValues[i]};
-            frobeniusNorm.emplace_back(::sqrt(
-                std::accumulate(eigenValues.begin(), eigenValues.end(), 0.,
-                                [](const double& a, const double& b) { return a + (b * b); })));
-        }
+    util::for_each_type<attributes::types3D>{}(
+        util::AddMetaDataProperties<3>{this, inport_.getData()});
 
-        tensorFieldOut_->addMetaData<FrobeniusNorm>(frobeniusNorm, TensorFeature::FrobeniusNorm);
+    /*
+    This is a bit of a hack but it'll do. What happens is that adding/removing of the default meta
+    data such as eigen values and eigen vectors is disabled by setting the respective properties to
+    readonly.
+    */
+    comp = getPropertiesByType<CompositeProperty>().front();
+    props = comp->getProperties();
+    for (size_t i{0}; i < 6; ++i) {
```

Comment:
Note: in `tensorfield2dmetadata.cpp` the test is `i < 4`. 
Consider using identifiers instead.

Reply:
Done.